src/App.tsx

```diff
@@ -16,80 +16,151 @@ export const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPHABET,
-  LENGTH,
+  LENGTH = 'length',
+  ALPHABETICAL = 'alphabetical',
+  NONE = '',
 }
 
-type ReorderOptions = {
+interface ReorderOptions {
   sortType: SortType,
   isReversed: boolean,
-};
+}
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
+export function getReorderedGoods(goods: string[],
+  { sortType, isReversed }: ReorderOptions) {
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABETICAL:
+      visibleGoods.sort((a: string, b: string) => {
+        if (a < b) {
+          return -1;
+        }
+
+        if (a > b) {
+          return 1;
+        }
+
+        return 0;
+      });
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((a: string, b: string) => {
+        if (a.length < b.length) {
+          return -1;
+        }
+
+        if (a.length > b.length) {
+          return 1;
+        }
+
+        return 0;
+      });
+      break;
 
-  return visibleGoods;
+    default:
+      break;
+  }
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean
+  sortType: SortType
+}
+
+export class App extends React.Component<{}, State> {
```

Comment:
There is a differenece between an empty object and null in typescript. In this case we use null to say that there aren't any props. 
```suggestion
export class App extends React.Component<null, State> {
```